api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -9,8 +22,54 @@
 @Service
 public class MailServiceImpl implements MailService {
 
+    private final Provider<MandrillApi> mandrillApiProvider;
+    private final Provider<CloudStorageService> cloudStorageServiceProvider;
+    private Provider<WorkbenchConfig> workbenchConfigProvider;
+
+    @Autowired
+    public MailServiceImpl(Provider<MandrillApi> mandrillApiProvider,
+                           Provider<CloudStorageService> cloudStorageServiceProvider,
+                           Provider<WorkbenchConfig> workbenchConfigProvider) {
+        this.mandrillApiProvider = mandrillApiProvider;
+        this.cloudStorageServiceProvider = cloudStorageServiceProvider;
+        this.workbenchConfigProvider = workbenchConfigProvider;
+    }
+
     public void send(Message msg) throws MessagingException {
         Transport.send(msg);
     }
 
+    public void sendWelcomeEmail(String contactEmail, String password, User user) throws MessagingException {
+        String apiKey = cloudStorageServiceProvider.get().readMandrillApiKey();
+        MandrillApiKeyAndMessage keyAndMessage = new MandrillApiKeyAndMessage();
+        keyAndMessage.setKey(apiKey);
+        MandrillMessage msg = buildWelcomeMessage(contactEmail, password, user);
+        keyAndMessage.setMessage(msg);
+        try {
+            MandrillMessageStatuses msgStatuses = mandrillApiProvider.get().send(keyAndMessage);
+            for (MandrillMessageStatus msgStatus : msgStatuses) {
+                if (msgStatus.getRejectReason() != null) {
+                  throw new MessagingException(msgStatus.getRejectReason());
+                }
+            }
+        } catch (ApiException e) {
+            throw new MessagingException("Sending email failed.");
+        } catch (MessagingException e) {
+            throw new MessagingException("Sending email failed with message: " + e.getMessage());
+        }
+    }
+
+    private MandrillMessage buildWelcomeMessage(String contactEmail, String password, User user) {
+        MandrillMessage msg = new MandrillMessage();
+        RecipientAddress toAddress = new RecipientAddress();
+        toAddress.setEmail(contactEmail);
+        msg.setTo(Collections.singletonList(toAddress));
+        String msgBody = "Your new account is: " + user.getPrimaryEmail() +
+          "\nThe password for your new account is: " + password;
```

Comment:
I think ideally the message body would contain a link to go log into workbench?

Reply:
This will be taken care of by Bru's PR that has the to spec welcome email.

Reply:
Do we want to start using that here? Or as a followup? (Probably makes sense to do a followup)

Reply:
Think it will be easier merge wise to merge this first and then plug in your email. (Plus actually have to do the other hooking up covered by 707.)